### PR TITLE
DGSPlanner bug fixes

### LINKED ISSUE #29164: Bad handling of incorrect inputs for DGSPlanner
Putting incorrect input in several places in DgsPlanner interface cause unhandled ValueError

### Steps to reproduce the behavior
Start DgsPlannerInterface. In the lattice parameters section start by putting `-` in one of the ux, uy, uz edits. Similarly, in the matrix UB widget, put a `-` and move away. It should just wait for good input, not pop up the error reporter. The same steps in MantidPlot are caught by the C++ layer, so you only have some messages in the log window.

### PR DESCRIPTION
**Pull request by AndreiSavici**
**Description of work.**

Fixed issues in DGSPlanner interface

**To test:**
Start the DGSPlanner interface, preferably in workbench. In the UB section, start typing `-1` for one of the u or v vector components. It should not crash. 

Fixes #29164

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/DGSPlanner/InstrumentSetupWidget.py [hunk 1]
```diff
@@ -95,25 +95,22 @@ def setData(self, index, value, role = QtCore.Qt.EditRole):
...
                 if column==0:
                     self.labels[row]=val
                 else:
                     self.dirstrings[row]=val
             elif column==2:
                 try:
-                    val=value.toInt()[0] #QVariant
-                except AttributeError:
-                    val=int(value) #string
+                    val=val=int(value)
```
Comment (DanielMurphy22) [inline scripts/DGSPlanner/InstrumentSetupWidget.py:105]: Should this line simply be:
```suggestion
                    val=int(value)
```

**Review by DanielMurphy22 — CHANGES_REQUESTED** [review]
The fix works well, and the coloring is great!

**Review by peterfpeterson — DISMISSED** [review]
As advertised: doesn't crash.

**Review by DanielMurphy22 — APPROVED** [review]
Well done!

**Pull request merged**